src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -383,6 +389,10 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 		return this._onDidStateChange.event;
 	}
 
+	public get onDidReconnectToTerminals(): Event<void> {
+		return this._onDidReconnectToTerminals.event;
+	}
```

Comment:
You can just use `this._onDidReconnectToTerminals.event` directly if you don't use outside this file:

```
this._register(this._onDidReconnectToTerminals.event(async () => await this._attemptTaskReconnection()));
```

Alternatively:

```
protected readonly _onDidReconnectToTerminalsEmitter: Emitter<void> = new Emitter();
protected readonly _onDidReconnectToTerminals = this._onDidReconnectToTerminalsEmitter.event;
```

Reply:
Doesn't look like this one was done yet

Reply:
this is actually not what we want to do because has to be accessible here - doing what is done for `onDidStateChange` https://github.com/microsoft/vscode/blob/6149b148b8e3e79af6f93d39ce11e108f8c841e9/src/vs/workbench/contrib/tasks/electron-sandbox/taskService.ts#L134-L151

Reply:
`protected` is probably fine though

Reply:
`protected` doesn't work actually

Reply:
no that doesn't work